components/webln.js

```diff
@@ -0,0 +1,106 @@
+import { createContext, createRef, useContext, useEffect, useImperativeHandle, useState } from 'react'
+
+const WebLNContext = createContext({})
+export const WebLNContextRef = createRef()
+
+const fetchWebLNProvider = async () => {
+  // sync provider from local storage
+  // TODO check if there is a better, more secure option
+  const configJson = window.localStorage.getItem('bc:config')
+  if (configJson) {
+    const WebLNProviders = await import('@getalby/bitcoin-connect').then((mod) => mod.WebLNProviders)
+    const config = JSON.parse(configJson)
+    if (config.connectorType === 'lnbits') return new WebLNProviders.LnbitsWebLNProvider(config.lnbitsInstanceUrl, config.lnbitsAdminKey)
+    // TODO implement other connector types
+  }
+}
+
+export function WebLNProvider ({ children }) {
+  const [provider, setProvider] = useState(null)
+  const [info, setInfo] = useState(null)
+  const [balance, setBalance] = useState(0)
+
+  const initProvider = async (provider) => {
+    const WebLNProviders = await import('@getalby/bitcoin-connect').then((mod) => mod.WebLNProviders)
```

Comment:
The only reason to cast to a specific provider type is if you need underlying functionality (such as being able to make extra Lnbits API calls not supported by WebLN). As far as I know this should not be needed

Reply:
Yep, also related to [this](https://github.com/stackernews/stacker.news/pull/715#discussion_r1437560662).